packages/ember-data/lib/system/model/model.js

```diff
@@ -1164,8 +1164,17 @@ var Model = Ember.Object.extend(Ember.Evented, {
   // This is a temporary solution until we refactor DS.Model to not
   // rely on the data property.
   willMergeMixin: function(props) {
-    Ember.assert('`data` is a reserved property name on DS.Model objects. Please choose a different property name for ' + this.constructor.toString(), !props.data);
-    Ember.assert('`store` is a reserved property name on DS.Model objects. Please choose a different property name for '+ this.constructor.toString(), !props.store);
+    var constructor = this.constructor;
+    [
+      'attributes', 'clientId', 'currentState', 'data', 'dirtyType',
+      'errors', 'fields', 'isDeleted', 'isDirty', 'isDestroyed',
+      'isDestroying', 'isEmpty', 'isError', 'isLoaded',
+      'isLoading', 'isNew', 'isReloading', 'isSaving', 'isValid',
+      'relatedTypes', 'relationshipNames', 'relationships',
+      'relationshipsByName', 'transformedAttributes', 'store'
+    ].forEach(function(reservedProperty) {
+      Ember.assert('`' + reservedProperty + '` is a reserved property name on DS.Model objects. Please choose a different property name for ' + constructor.toString(), !props[reservedProperty]);
```

Comment:
Hey, can we actually refactor this, to instead of iterating, does something like Ember.assert( intersection(props, reservedProps) so the whole method call will get stripped in prod